tools/clang/test/DXC/Passes/ScalarReplHLSL/hitobject_make_scalarrepl.ll

```diff
@@ -0,0 +1,130 @@
+; RUN: %dxopt %s -hlsl-passes-resume -scalarrepl-param-hlsl -S | FileCheck %s
+; REQUIRES: dxil-1-9
+
+;
+; Buffer Definitions:
+;
+; cbuffer $Globals
+; {
+;
+;   [0 x i8] (type annotation not present)
+;
+; }
+;
+;
+; Resource Bindings:
+;
+; Name                                 Type  Format         Dim      ID      HLSL Bind  Count
+; ------------------------------ ---------- ------- ----------- ------- -------------- ------
+; $Globals                          cbuffer      NA          NA     CB0   cb4294967295     1
+;
+target datalayout = "e-m:e-p:32:32-i1:32-i8:32-i16:32-i32:32-i64:64-f16:32-f32:32-f64:64-n8:16:32:64"
+target triple = "dxil-ms-dx"
+
+%ConstantBuffer = type opaque
+%dx.types.HitObject = type { i8* }
+%"class.dx::HitObject" = type { i32 }
+%struct.RayDesc = type { <3 x float>, float, <3 x float>, float }
+
+@"$Globals" = external constant %ConstantBuffer
+
+; Function Attrs: nounwind
+define void @"\01?main@@YAXXZ"() #0 {
+entry:
+  %hit = alloca %dx.types.HitObject, align 4
+  %tmp = alloca %dx.types.HitObject, align 4
+  %ray = alloca %struct.RayDesc, align 4
+  %tmp2 = alloca %dx.types.HitObject, align 4
+  %0 = bitcast %dx.types.HitObject* %hit to i8*, !dbg !23 ; line:42 col:3
+  call void @llvm.lifetime.start(i64 4, i8* %0) #0, !dbg !23 ; line:42 col:3
+; CHECK:  %{{[^ ]+}} = call %dx.types.HitObject* @"dx.hl.op..%dx.types.HitObject* (i32, %dx.types.HitObject*)"(i32 358, %dx.types.HitObject* %{{[^ ]+}})
```

Comment:
This match would succeed even if the IR is unchanged from with is in the test case now, so we probably need some more specific matches, maybe a variable match in order to verify that this IR is actually being transformed.

Reply:
That is because this call to `HitObject_MakeNop` is not expected to change. SroaHLSL must only flatten the `RayDesc` param in the `DxHitObject_MakeMiss` call below and the test is checking for exactly that.
@llvm-beanz If that is not what we want, could you provide more specific guidance of what you want to see changed?

Reply:
If the goal is to make sure this doesn't get scalarized, we should have some variable matches to make sure that the values move as expected through the operations.

Reply:
Added explicit `CHECK`s that `RayDesc` gets scalarized and all other params remain in place. Also checking that the ptr from the default ctr remains unused.
@llvm-beans does this resolve this request?

Reply:
Asserting that `CI->use_empty()` as discussed offline. Resolved